linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -2991,8 +2992,20 @@ private[emitter] class FunctionEmitter(sjsGen: SJSGen) {
       js.Apply(globalVar(field, (className, method.name)), args)
     }
 
+    private def genCallBuiltinOrPolyfillHelper(builtinESVersion: ESVersion,
```

Comment:
While it is nice to see more structure here, this "models" the `builtinESVersion` as a property of the callsite, rather than a property of the builtin.

This might be a chance to abstract over this? This might even simplify the CoreJSLib, rather than collecting the specific helpers that are necessary, it could just loop over all known ones and generate if necessary.

Reply:
I added a commit that unifies the polyfillable builtins as a single abstraction that knows where to find it and in which version. It may be overkill, but maybe not? WDYT?